### DISCUSSION TITLE
[SOLVED] How to test the _createFn callback that is inside the provider

### DISCUSSION THREAD

**Discussion by astrid-dev** [rrousselGit/riverpod#2100] [Q&A] [answered]
when running `flutter test --coverage` I always get the variables that contains my providers as not tested and the docs only information about testing is overriding to mock the providers but not testing the function.

How can I test that?

Example:

using the counter basic flutter app (the one that is created with `flutter create`)
and replacing the logic to use these `StateProvider`s, then creating the test for the screen (overriding the providers)
and running `flutter test --coverage` the variables that contain the providers are marked as not tested, which makes sense because I used an override, so the question is how can I test the original so the coverage know they are tested
```dart
// providers
import 'package:flutter_riverpod/flutter_riverpod.dart';

final counterProvider = StateProvider((ref) => 0);

final counterStringProvider = StateProvider((ref) {
  final count = ref.watch(counterProvider);
  return count.toString();
});
```
coverage result
<img width="583" alt="Screen Shot 2023-01-19 at 16 49 41" src="https://user-images.githubusercontent.com/6403537/213579644-d768b54b-1d51-485d-9db2-97eee32a4d68.png">

or is a good practice to ignore them for the coverage?

**Comment by rrousselGit** [comment]
I don't understand. You could highlight exactly what's not tested?

**Reply by astrid-dev** [reply]
yeah, I just updated the question with an example

**Comment by rrousselGit** [comment, marked as answer]
As you've said, you've used an override. So your provider function is not tested

You need to test the function too, by executing the provider without overriding it.

**Reply by astrid-dev** [reply]
is there a way to do that without testing a widget? asking because I will not have the `ref` without the provider scope

**Reply by danielmahon** [reply]
I think you can test `counterStringProvider` by overriding `counterProvider` with a mock and then you don't need to mock anything to test `counterProvider`.

**Reply by rrousselGit** [reply]
You can freely read providers using ProviderContainer

**Reply by rrousselGit** [reply]
You can freely read providers using ProviderContainer

**Reply by danielmahon** [reply]
@rurickdev  https://docs-v2.riverpod.dev/docs/cookbooks/testing Look at the "dart only" examples

**Reply by astrid-dev** [reply]
Thanks! I completely miss the Dart Only tab in the examples, I finished my tests with that.
